pkg/agent/containerd/containerd.go

```diff
@@ -110,15 +112,160 @@ func Run(ctx context.Context, cfg *config.Node) error {
 	return PreloadImages(ctx, cfg)
 }
 
+// Watcher is a controller that watch the agent/images folder
+// to ensure that every new file is added to the watcher state
+func Watcher(ctx context.Context, cfg *config.Node) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logrus.Errorf("Error to create a watcher: %s", err.Error())
+		return
+	}
+
+	// Add agent/images path to the watcher.
+	err = watcher.Add(cfg.Images)
+	if err != nil {
+		logrus.Errorf("Error when creating the watcher controller: %s", err.Error())
+		return
+	}
+
+	client, err := Client(cfg.Containerd.Address)
+	if err != nil {
+		logrus.Errorf("Error to create containerd client: %s", err.Error())
+		return
+	}
+
+	criConn, err := cri.Connection(ctx, cfg.Containerd.Address)
+	if err != nil {
+		logrus.Errorf("Error to create CRI connection: %s", err.Error())
+		return
+	}
+
+	imageClient := runtimeapi.NewImageServiceClient(criConn)
+
+	defer watcher.Close()
+	defer client.Close()
+	defer criConn.Close()
+
+	fileInfos, err := os.ReadDir(cfg.Images)
+	if err != nil {
+		logrus.Errorf("Unable to read images in %s: %v", cfg.Images, err)
+		return
+	}
+
+	// Ensure that our images are imported into the correct namespace
+	ctx = namespaces.WithNamespace(ctx, constants.K8sContainerdNamespace)
+
+	// At startup all leases from k3s are cleared; we no longer use leases to lock content
+	if err := clearLeases(ctx, client); err != nil {
+		logrus.Errorf("Error while clearing leases: %s", err.Error())
+		return
+	}
+
+	// Clear the pinned labels on all images previously pinned by k3s
+	if err := clearLabels(ctx, client); err != nil {
+		logrus.Errorf("Error while clearing labes: %s", err.Error())
+		return
+	}
+
+	// create the state that watcher will watch
+	stateFileInfos := make(map[string]fs.FileInfo)
+	for _, dirEntry := range fileInfos {
+		if dirEntry.IsDir() {
+			continue
+		}
+
+		// get the file info to add to the state map
+		fileInfo, err := dirEntry.Info()
+		if err != nil {
+			logrus.Errorf("Error while getting the info from file: %s", err.Error())
```

Comment:
Convention is to use %v for errors, and just pass in the error itself.
```suggestion
			logrus.Errorf("Error while getting the info from file: %v", err)
```